PITCHME.md

```diff
@@ -0,0 +1,530 @@
+- rosdoc => doxygen made easy
+- rostest => gtest made easy
+- ROSTCP => networking made easy
+- rosmsg => serialization made easy
+- etc.
+
++++
+
+A set of components for robots
+------------------------------
+
+- calibration tools
+- development tools
+- visualisaton
+- navigation
+- manipulation
+- ...
+
++++
+
+Big community
+-------------
+
+- ROS wiki page views : 1,394,363
+- International : US, CN, JP, DE > 25,000
+- ROS wiki users : 5875
+- ROS pkg downloads : 8,441,279 from 113,000 IPs
+- ROS research papers : 2683
+- ROS Robots : > 100 (http://wiki.ros.org/Robots)
+
+source : http://wiki.ros.org/Metrics
+
++++
+
+Permissive licensing
+--------------------
+
+- ROS core BSD license (Free to use, modify, sell)
+- ROS tools: Open Source/Free licenses (free to use)
+
+
+
+---
+
+ROS1 Basics
+===========
+
+## Do the tutorials !
+
+- Environment and Tools
+- Packages and Dependencies
+- Launchers and Parameters
+- Messages, Topics and Services
+
++++
+
+ROS1 Environment
+----------------
+
+`$ source setup.bash`
+- modify your current SHELL (`bash`) environment.
+- <span style="color:green">GOOD</span> : easy to modify quickly (`export MY_ROS_ENVVAR=42`)
+- <span style="color:red">BAD</span> : hard to debug (`ps aux | grep my_ros_program`) => which environment configuration ?
+
++++
+
+ROS1 Commands
+-------------
+
+`rosls`, `roscd`, `rospack`, etc.
+- allows easy navigation in ROS environment
+- <span style="color:green">GOOD</span> : easy to play around in your ROS environment
+- <span style="color:green">GOOD</span> : allow plug-n-play for node / comm checking
+- <span style="color:red">BAD</span> : not fully isolated from the rest of the system
+
++++
+
+ROS1 Packages
+-------------
+
+`package.xml` and `CMakeLists.txt`
+- describes a ROS package
+- <span style="color:green">GOOD</span> : modular compilable source code (`catkin_make` or `catkin build`)
+- <span style="color:green">GOOD</span> : modular interactive runtime (ROS API)
+- <span style="color:red">BAD</span> : not easy to compose, easy to shoot yourself
+- <span style="color:green">GOOD</span> : integrating with your package manager
+- <span style="color:red">BAD</span> : not quite integrated, it goes into `/opt/ros/<rosdistro>`
+
++++
+
+ROS1 Dependencies
+-----------------
+
+`rosdep resolve depkey`
+
+- enable cross-platform package management
+- <span style="color:green">GOOD</span> : can edit github's `ros/rosdistro` repository to add dependencies 
+- <span style="color:red">BAD</span> : currently can confuse ROS packages and dependencies (BUG)
+
++++
+
+ROS1 Nodes
+----------
+
+`rospy.init_node()` or `ros::init()`
+- declares the current process as a ROS node
+- <span style="color:green">GOOD</span> : one node match one process
+- <span style="color:red">BAD</span> : special `rosmaster`, `rosout` nodes
+- <span style="color:red">BAD</span> : pollute your process global state
+- <span style="color:red">BAD</span> : async programming can be VERY confusing
```

Comment:
Maybe mention multi-threaded spinner?

Reply:
Not sure what you mean or maybe I just dont know what it is...